public/js/lean.js

```diff
@@ -1,41 +1,96 @@
+  if (googleApiKey) {
+    var scriptSrc = 'https://maps.googleapis.com/maps/api/js?key=' + googleApiKey + '&libraries=places';
+    var alreadyLoaded = document.querySelector('script[src="' + scriptSrc + '"]');
+    if (!alreadyLoaded) {
+      var googleScript = document.createElement('script');
+      googleScript.setAttribute('src', scriptSrc);
+      document.head.appendChild(googleScript);
+    }
 
-    var countries = _.map(widget.countries, function (countryObject) {
-      return countryObject.country;
-    });
+    var input = el.querySelector('input[name=' + widget.fieldName + ']');
+    input.addEventListener('input', function () {
+      var google = window.google;
 
-    var searchOptions = {
-      componentRestrictions: { country: countries }
-    };
+      var countries = _.map(widget.countries, function (countryObject) {
+        return countryObject.country;
+      });
 
-    if (widget.types) {
-      searchOptions.types = [widget.types];
-    }
-    if (widget.origin) {
-      searchOptions.origin = new google.maps.LatLng(widget.originLatitude, widget.originLongitude);
-    }
-    if (widget.bounds) {
-      searchOptions.bounds = new google.maps.LatLngBounds(
-        new google.maps.LatLng(widget.northEastLatitude, widget.northEastLongitude),
-        new google.maps.LatLng(widget.southWestLatitude, widget.southWestLongitude)
-      );
-
-      if (widget.strictBounds) {
-        searchOptions.strictBounds = true;
+      var searchOptions = {
+        componentRestrictions: { country: countries }
+      };
+
+      if (widget.types) {
+        searchOptions.types = [widget.types];
+      }
+
+      if (widget.origin) {
+        searchOptions.origin = new google.maps.LatLng(widget.originLatitude, widget.originLongitude);
       }
-    }
 
-    new google.maps.places.Autocomplete(input, searchOptions); // eslint-disable-line no-new
-  });
+      if (widget.bounds) {
+        searchOptions.bounds = new google.maps.LatLngBounds(
+          new google.maps.LatLng(widget.northEastLatitude, widget.northEastLongitude),
+          new google.maps.LatLng(widget.southWestLatitude, widget.southWestLongitude)
+        );
+
+        if (widget.strictBounds) {
+          searchOptions.strictBounds = true;
+        }
+      }
+
+      var autocomplete = new google.maps.places.Autocomplete(input, searchOptions); // eslint-disable-line no-new
+      autocomplete.setFields(['address_component']);
+
+      if (widget.splitAddress && widget.addressParts && widget.addressParts.length) {
+        var inputNames;
+
+        autocomplete.addListener('place_changed', function () {
+          inputNames = {};
+          var split = widget.addressParts;
+          var place = autocomplete.getPlace();
+
+          if (place.address_components) {
+            for (var i = 0; i < split.length; i++) {
+              var element = el.querySelector('.apos-forms-input-' + split[i]);
```

Comment:
Classes should only be used for visual purposes, use a data attribute to identify the subfield here.

Reply:
Done